File: leetcodes/20. Valid Parentheses/validParenthese.java
class Solution {
    public boolean isValid(String s) {
      int len = s.length();
      if (len %2 != 0) return false;
      //create stack
      Stack<Character> temp = new Stack<Character>();
      for (char c : s.toCharArray()){
        if (c == '(') {
          temp.push(')');
        } else if (c == '[') {
          temp.push(']');
        } else if (c == '{') {
          temp.push('}');
        } else if ( temp.isEmpty()|| temp.pop() != c) return false;
      }
      return temp.isEmpty();
    }
}

File: leetcodes/20. Valid Parentheses/validParenthese.cs
public class Solution {
    public bool IsValid(string s) {
      //dictionary -> O(n) space -> O(n)
      //stack -> O(n) space O(n)
      //check len of s if it's odd -> false
      if (s.Length % 2 != 0) return false;
         Stack <char> stack = new Stack<char>();
      foreach (var c in s) {
        if (c == '(') stack.Push(')');
        else if (c =='[') stack.Push(']');
        else if (c =='{') stack.Push('}');
        else if (stack.Count == 0 || stack.Pop() != c) return false;
      }
      return stack.Count == 0;
    }
}